Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -815,6 +842,35 @@ void qSlicerMarkupsModuleWidget::updateWidgetFromMRML()
     {
     this->updateRow(m);
     }
+  
+  //check for active curve and enable 
+  vtkMRMLSelectionNode *selNode = d->selectionNode();
+  if (selNode)
+  {
+	  vtkMRMLNode *mrmlNode = this->mrmlScene()->GetNodeByID(selNode->GetActivePlaceNodeID());
+	  vtkMRMLMarkupsCurveNode *markupsCurveNode = vtkMRMLMarkupsCurveNode::SafeDownCast(mrmlNode);
+	  if (markupsCurveNode)
+	  {
+		  d->applyCurveResamplingPushButton->setEnabled(true);
+	  }
+	  else
+	  {
+		vtkMRMLMarkupsClosedCurveNode *markupsClosedCurveNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(mrmlNode);
+		if (markupsClosedCurveNode)
+		{
+			d->applyCurveResamplingPushButton->setEnabled(true);
+		}
+		else
+		{
+			d->applyCurveResamplingPushButton->setEnabled(false);
+		}
+	  }
+	 
+  }
+  else
+  {
+	  d->applyCurveResamplingPushButton->setEnabled(false);
```

Comment:
We should offer resample for all markup nodes, just disable the number of points spinbox. It would allow us to snap any annotation to a model.

Reply:
Could we add resample for other markup node types at a later point? We'd like to get this working for curves as soon as we can.

Reply:
Yes